src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -823,6 +840,24 @@ static string GetStyledText(TaggedText taggedText, bool isInCodeBlock)
             }
         }
 
+        public static LanguageInformation? GetLanguageInformation(LSP.TextDocumentItem textDocument)
+        {
+            if (s_extensionToLanguageInformation.TryGetValue(Path.GetExtension(textDocument.Uri.AbsolutePath), out var languageInformation))
+            {
+                return languageInformation;
+            }
+
+            // It is totally possible to not find language based on the file path (e.g. a newly created file that hasn't been saved to disk).
+            // In that case, we use the languageId that the client gave us.
+            var languageId = textDocument.LanguageId;
+            return languageId switch
```

Comment:
Rather than this map having to exist, should we just be holding into the language ID as a string and let the miscellaneous files workspace stuff deal with this mapping? It seems a bit odd we're otherwise special-casing C# and VB this far down into the system.

Reply:
done!